pkg/controller/seed-controller-manager/default-application-controller/controller_test.go

```diff
@@ -389,6 +392,31 @@ func TestReconcile(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name:                        "scenario 13: should create default application in cluster with ready Cilium system application in configured default applicationinstallation namespace",
+			defaultApplicationNamespace: applicationInstallationNamespace,
+			cluster:                     genCluster(clusterName, defaultDatacenterName, false, ciliumCNISettings),
+			systemAppInstallationValues: map[string]any{"status": "ready"},
```

Comment:
I think this test is independent of the Cilium and the systemAppInstallationValues (especially since we only mock a Cilium state in there as well) and the state has nothing to do with where the AppInstall is synched

Reply:
There is a [condition](https://github.com/kubermatic/kubermatic/blob/main/pkg/controller/seed-controller-manager/default-application-controller/controller.go#L166-L173) for checking the status of cilium app when the cni is managed via application before generating and deploying the default applications. The condition is only checking status field.